rhombus-lib/rhombus/private/amalgam/path-object.rkt

```diff
@@ -36,7 +45,8 @@
 (define-primitive-class Path path
   #:lift-declaration
   #:no-constructor-static-info
-  #:instance-static-info ((#%compare ((< path<?)
+  #:instance-static-info ((#%append Path.extend)
```

Comment:
I think `Path.extend` doesn’t behave exactly like an `append` implementation and thus `++` shouldn’t work.  In general, an `append` method receives arguments that inherit the same `append` implementation, but `Path.extend` can also receive path strings and the two special symbols `'up` and `'same`.  Maybe a separate `Path.append` that only works with path values could be defined, if that seems useful (but `Appendable` then needs to be updated to handle path values).

Reply:
If we don't like `++`, I favor Python's trick of `dir / file` (overloading division) as a natural way to write paths.

Reply:
Initially I wanted to use `/` or `//` as a dedicated operator (like Python.). The problem with `/` is that it's complicated and probably the wrong choice since there really isn't any existing overloading of the arithmetic operators.  `//` is right out since it is a comment token at the shrubbery level.

I agree `++` is probably not the right place to do this.  I do think it would be nice if there was an infix operator to do this.  Maybe adding `Path./` operator is a good idea?

Reply:
Would `+/` work well if it were allowed as an operator? I think the prohibition against `/` at the end of an operator could be lifted by adding the narrower constraint that an operator ending in `/` cannot be followed by `/` or `*`.

Reply:
Removed support for `Appendable` and add the `+/` operator on `PathString`s